2018-11-02  Mudith Mallajosyula  <mudithm@g.ucla.edu>

* timespec.h (timespec_cmp): use comparison operators instead of subtraction
  when comparing values to avoid integer overflow issues

--- coreutils-with-bug/lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ coreutils-with-bug-new/lib/timespec.h	2018-11-02 16:25:15.811087000 -0700
@@ -45,8 +45,19 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  
+  if (a.tv_sec > b.tv_sec)
+    return 1;
+  else if (a.tv_sec < b.tv_sec)
+    return -1;
+  else
+    if (a.tv_nsec > b.tv_nsec)
+      return 1;
+    else if (a.tv_nsec < b.tv_nsec)
+      return -1;
+    else
+      return 0;
+
 }
 
 # if ! HAVE_DECL_NANOSLEEP
